app/src/main/kotlin/io/homeassistant/companion/android/controls/HaControlsProviderService.kt

```diff
@@ -147,7 +147,8 @@ class HaControlsProviderService : ControlsProviderService() {
                                     entityId = entity.entityId,
                                     serverId = serverId,
                                     serverName = serverNames[serverId],
-                                    area = getAreaForEntity(entity.entityId, serverId)
+                                    area = getAreaForEntity(entity.entityId, serverId),
+                                    splitMultiServerIntoStructure = splitMultiServersIntoStructures()
```

Comment:
You only need to fetch this value once as it is the same for all entities/servers. Very minor performance improvement but can this be placed outside the loop?